### PR TITLE
Allow sortmerna to operate in indexing-only mode

### PR DESCRIPTION
**Pull request by pinin4fjords**
Where we use sortmerna (rnaseq, riboseq, subworkflows/preprocess_rnaseq), we do so wastefully. By default, the tool indexes every time, which can be time-consuming. sortmerna does, however, have a an option `--index` by which the indexing can be turned off (`--index 0`) or be made the only thing that happens (`--index 1`). 

In this PR I propose changes to allow use of sortmerna in indexing-only mode:

- All outputs made optional.
- Argument wrangling before `script:` customised to allow the above.
- `subworkflows/preprocess_rnaseq` customised to only run indexing once.

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/sortmerna/tests/indexing_only.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (maxulysse) [inline modules/nf-core/sortmerna/tests/indexing_only.config:1]:
```suggestion
```

File: modules/nf-core/sortmerna/tests/premade_index.config [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (maxulysse) [inline modules/nf-core/sortmerna/tests/premade_index.config:1]:
```suggestion
```

**Review by maxulysse — APPROVED** [review]
Looking good

**Comment by pinin4fjords** [comment]
Thanks @maxulysse !

**Comment by GallVp** [comment]
Thank you @pinin4fjords

Really nice work!

**Pull request merged**